wcfsetup/install/files/ts/WoltLabSuite/Core/Media/Editor.ts

```diff
@@ -0,0 +1,419 @@
+  }
+
+  public _ajaxSetup(): ReturnType<AjaxCallbackSetup> {
+    return {
+      data: {
+        actionName: "update",
+        className: "wcf\\data\\media\\MediaAction",
+      },
+    };
+  }
+
+  public _ajaxSuccess(): void {
+    UiNotification.show();
+
+    if (this._callbackObject._editorSuccess) {
+      this._callbackObject._editorSuccess(this._media, this._oldCategoryId);
+      this._oldCategoryId = 0;
+    }
+
+    UiDialog.close(`mediaEditor_${this._media!.mediaID}`);
+
+    this._media = null;
+  }
+
+  /**
+   * Is called if an editor is manually closed by the user.
+   */
+  protected _close(): void {
+    this._media = null;
+
+    if (this._callbackObject._editorClose) {
+      this._callbackObject._editorClose();
+    }
+  }
+
+  /**
+   * Initializes the editor dialog.
+   *
+   * @since 5.3
+   */
+  protected _initEditor(content: HTMLElement, data: InitEditorData): void {
+    this._availableLanguageCount = ~~data.returnValues.availableLanguageCount;
+    this._categoryIds = data.returnValues.categoryIDs.map((number) => number);
+
+    if (data.returnValues.mediaData) {
+      this._media = data.returnValues.mediaData;
+    }
+
+    // make sure that the language chooser is initialized first
+    setTimeout(() => {
+      if (this._availableLanguageCount > 1) {
+        LanguageChooser.setLanguageId(
+          `mediaEditor_${this._media!.mediaID}_languageID`,
+          this._media!.languageID || window.LANGUAGE_ID,
+        );
+      }
+
+      if (this._categoryIds.length) {
+        const categoryID = content.querySelector("select[name=categoryID]") as HTMLSelectElement;
+        if (this._media!.categoryID) {
+          categoryID.value = this._media!.categoryID.toString();
+        } else {
+          categoryID.value = "0";
+        }
+      }
+
+      const title = content.querySelector("input[name=title]") as HTMLInputElement;
+      const altText = content.querySelector("input[name=altText]") as HTMLInputElement;
+      const caption = content.querySelector("textarea[name=caption]") as HTMLInputElement;
+      const mediaId = this._media!.mediaID;
+
+      if (this._availableLanguageCount > 1 && this._media!.isMultilingual) {
+        if (document.getElementById(`altText_${mediaId}`)) {
+          LanguageInput.setValues(`altText_${this._media!.mediaID}`, (this._media!.altText || {}) as I18nValues);
+        }
+
+        if (document.getElementById(`caption_${mediaId}`)) {
+          LanguageInput.setValues(`caption_${this._media!.mediaID}`, (this._media!.caption || {}) as I18nValues);
+        }
+
+        LanguageInput.setValues(`title_${this._media!.mediaID}`, (this._media!.title || {}) as I18nValues);
+      } else {
+        title.value = this._media!.title ? this._media!.title[this._media!.languageID || window.LANGUAGE_ID] : "";
+        if (altText) {
+          altText.value = this._media!.altText
+            ? this._media!.altText[this._media!.languageID || window.LANGUAGE_ID]
+            : "";
```

Comment:
This could be simplified to:

`altText.value = media.altText?.[media.languageID || window.LANGUAGE_ID] || ""`

(note the `?.` operator)